### PR TITLE
Compile WAMR on platforms that don't support IPV6

### PR DESCRIPTION
**Pull request by guangyuhu**
### Summary

Adding changes so that WAMR can compile on platforms that don't have IPV6 support.

### CODE REVIEW

File: core/shared/platform/common/posix/posix_socket.c [hunk 1]
```diff
@@ -168,9 +176,15 @@ os_socket_bind(bh_socket_t socket, const char *host, int *port)
...
-    *port = ntohs(addr.ss_family == AF_INET
-                      ? ((struct sockaddr_in *)&addr)->sin_port
-                      : ((struct sockaddr_in6 *)&addr)->sin6_port);
+    if (addr.ss_family == AF_INET) {
+        *port = ntohs(((struct sockaddr_in *)&addr)->sin_port);
+    } else {
+#ifdef IPPROTO_IPV6
+        *port = ntohs(((struct sockaddr_in6 *)&addr)->sin6_port);
+#else
+    goto fail;
```
Comment (wenyongh) [inline core/shared/platform/common/posix/posix_socket.c:185]: Coding style, should be 8 leading spaces

File: core/shared/platform/common/posix/posix_socket.c [hunk 2]
```diff
@@ -881,15 +913,23 @@ os_socket_get_ip_multicast_ttl(bh_socket_t socket, uint8_t *ttl_s)
 int
 os_socket_set_ipv6_only(bh_socket_t socket, bool is_enabled)
 {
+#ifdef IPPROTO_IPV6
     return os_socket_setbooloption(socket, IPPROTO_IPV6, IPV6_V6ONLY,
                                    is_enabled);
+#else
+        return BHT_ERROR;
```
Comment (wenyongh) [inline core/shared/platform/common/posix/posix_socket.c:920]: Should be 4 leading spaces

File: core/shared/platform/common/posix/posix_socket.c [hunk 3]
```diff
@@ -881,15 +913,23 @@ os_socket_get_ip_multicast_ttl(bh_socket_t socket, uint8_t *ttl_s)
...
 }
 
 int
 os_socket_get_ipv6_only(bh_socket_t socket, bool *is_enabled)
 {
+#ifdef IPPROTO_IPV6
     return os_socket_getbooloption(socket, IPPROTO_IPV6, IPV6_V6ONLY,
                                    is_enabled);
+#else
+        return BHT_ERROR;
```
Comment (wenyongh) [inline core/shared/platform/common/posix/posix_socket.c:931]: Same as above

**Review by wenyongh — COMMENTED** [review]
Hi, there are some coding style issues, please help to resolve, or refer to:
https://github.com/bytecodealliance/wasm-micro-runtime/actions/runs/3549594148/jobs/5962153784

```
    1. Install clang-format-12.0.0
    Normally we can install it by `sudo apt-get install clang-format-12`,
    or download the `clang+llvm-12.0.0-xxx-tar.xz` package from
      https://github.com/llvm/llvm-project/releases/tag/llvmorg-12.0.0
    and install it

    2. Format the C/C++ source file
    ``` shell
    cd path/to/wamr/root
    clang-format-12 --style file -i path/to/file
    ```
```

**Comment by guangyuhu** [comment]
Thanks for your comments, ran clang-format over the changes so the style issues should be resolved now.

File: core/shared/platform/common/posix/posix_socket.c [hunk 4]
```diff
@@ -168,9 +176,16 @@ os_socket_bind(bh_socket_t socket, const char *host, int *port)
...
     }
 
-    *port = ntohs(addr.ss_family == AF_INET
-                      ? ((struct sockaddr_in *)&addr)->sin_port
-                      : ((struct sockaddr_in6 *)&addr)->sin6_port);
+    if (addr.ss_family == AF_INET) {
+        *port = ntohs(((struct sockaddr_in *)&addr)->sin_port);
+    }
+    else {
+#ifdef IPPROTO_IPV6
```
Comment (loganek) [inline core/shared/platform/common/posix/posix_socket.c:183]: I'm not sure if we ever actually get to the `else` branch, I'd expect the `bind()` to fail sooner if IPv6 is not supported.

I think we can probably fail way before making any syscall if IPv6 is not supported; my suggestion would be to move textual_addr_to_sockaddr at the very top, and move this if...else just after that. What do you think?

File: core/shared/platform/common/posix/posix_socket.c [hunk 5]
```diff
@@ -318,6 +333,7 @@ os_socket_inet_network(bool is_ipv4, const char *cp, bh_ip_addr_buffer_t *out)
         /* Note: ntohl(INADDR_NONE) == INADDR_NONE */
         out->ipv4 = ntohl(out->ipv4);
     }
+#ifdef IPPROTO_IPV6
     else {
```
Comment (loganek) [inline core/shared/platform/common/posix/posix_socket.c:337]: Do we handle it correctly? I think if `is_ipv4` is set to `false` and ipv6 is not supported, we should probably return an error instead (I think right now we return `BHT_OK`).

File: core/shared/platform/common/posix/posix_socket.c (hunk 4 above)
Comment (guangyuhu) [reply core/shared/platform/common/posix/posix_socket.c]: Makes sense, moved these bits to the start of this function.

File: core/shared/platform/common/posix/posix_socket.c (hunk 5 above)
Comment (guangyuhu) [reply core/shared/platform/common/posix/posix_socket.c:336]: You are right, changed this to return `BHT_ERROR` and set `EAFNOSUPPORT` as the `errno` where applicable.

**Review by loganek — APPROVED** [review]
LGTM

**Pull request merged**